components/brave_wallet/browser/asset_discovery_manager.cc

```diff
@@ -170,228 +176,174 @@ void AssetDiscoveryManager::OnGetSolanaTokenRegistry(
+                         triggered_by_accounts_added);
 }
 
 void AssetDiscoveryManager::DiscoverEthAssets(
-    const std::string& chain_id,
-    mojom::CoinType coin,
     const std::vector<std::string>& account_addresses,
-    bool triggered_by_accounts_added,
-    const std::string& from_block,
-    const std::string& to_block) {
+    bool triggered_by_accounts_added) {
   if (account_addresses.empty()) {
-    CompleteDiscoverAssets(
-        chain_id, std::vector<mojom::BlockchainTokenPtr>(),
-        mojom::ProviderError::kInvalidParams,
-        l10n_util::GetStringUTF8(IDS_WALLET_INVALID_PARAMETERS),
-        triggered_by_accounts_added);
-    return;
-  }
-
-  // Asset discovery only supported on select EVM chains
-  if (coin != mojom::CoinType::ETH ||
-      !base::Contains(GetAssetDiscoverySupportedEthChains(), chain_id)) {
-    CompleteDiscoverAssets(
-        chain_id, std::vector<mojom::BlockchainTokenPtr>(),
-        mojom::ProviderError::kMethodNotSupported,
-        l10n_util::GetStringUTF8(IDS_WALLET_METHOD_NOT_SUPPORTED_ERROR),
-        triggered_by_accounts_added);
-    return;
-  }
-
-  // Asset discovery only supported when using Infura proxy
-  GURL infura_url = GetInfuraURLForKnownChainId(chain_id);
-  GURL active_url = GetNetworkURL(prefs_, chain_id, coin);
-  if (infura_url.host() != active_url.host()) {
-    CompleteDiscoverAssets(
-        chain_id, std::vector<mojom::BlockchainTokenPtr>(),
-        mojom::ProviderError::kMethodNotSupported,
-        l10n_util::GetStringUTF8(IDS_WALLET_METHOD_NOT_SUPPORTED_ERROR),
-        triggered_by_accounts_added);
+    CompleteDiscoverAssets({}, triggered_by_accounts_added);
     return;
   }
 
-  for (const auto& account_address : account_addresses) {
-    if (!EthAddress::IsValidAddress(account_address)) {
-      CompleteDiscoverAssets(
-          chain_id, std::vector<mojom::BlockchainTokenPtr>(),
-          mojom::ProviderError::kInvalidParams,
-          l10n_util::GetStringUTF8(IDS_WALLET_INVALID_PARAMETERS),
-          triggered_by_accounts_added);
-      return;
-    }
-  }
-
   std::vector<mojom::BlockchainTokenPtr> user_assets =
-      BraveWalletService::GetUserAssets(chain_id, mojom::CoinType::ETH, prefs_);
+      BraveWalletService::GetUserAssets(prefs_);
   auto internal_callback =
       base::BindOnce(&AssetDiscoveryManager::OnGetEthTokenRegistry,
-                     weak_ptr_factory_.GetWeakPtr(), chain_id,
-                     account_addresses, std::move(user_assets),
-                     triggered_by_accounts_added, from_block, to_block);
+                     weak_ptr_factory_.GetWeakPtr(), account_addresses,
+                     std::move(user_assets), triggered_by_accounts_added);
 
-  BlockchainRegistry::GetInstance()->GetAllTokens(
-      chain_id, mojom::CoinType::ETH, std::move(internal_callback));
+  BlockchainRegistry::GetInstance()->GetEthTokenListMap(
+      GetAssetDiscoverySupportedEthChains(), std::move(internal_callback));
```

Comment:
I think it could return the map directly.  I was just trying to keep things consistent with the other functions in the BlockchainRegistry class, but upon reading your comment I see that those take a callback because they are a part of the mojo interface, when this is not. I will update.

Reply:
I was mostly following the pattern of the other BlockchainRegistry functions, but after reading your comment I see that they accept a callback parameter because that are a part of the mojo interface definition.  I will update to remove the callaback parameter.